python/pyarrow/tests/test_table.py

```diff
@@ -271,7 +271,9 @@ def ne(xarrs, yarrs):
     eq([a, c], [d])
     ne([c, a], [a, c])
 
-    assert not pa.chunked_array([], type=pa.int32()).equals(None)
+    # ARROW-4822
+    with pytest.raises(AttributeError):
+        pa.chunked_array([], type=pa.int32()).equals(None)
```

Comment:
This is a behaviour change though.

Before:
```
>>> import pyarrow
>>> table = pyarrow.chunked_array([], type=pyarrow.int32())
>>> table is None
False
>>> table == None
False
```

After:

```
>>> import pyarrow
>>> table = pyarrow.chunked_array([], type=pyarrow.int32())
>>> table is None
False
>>> table == None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/table.pxi", line 187, in pyarrow.lib.ChunkedArray.__eq__
    return self.equals(other)
  File "pyarrow/table.pxi", line 212, in pyarrow.lib.ChunkedArray.equals
    CChunkedArray* other_arr = other.chunked_array
AttributeError: 'NoneType' object has no attribute 'chunked_array'
```

Reply:
Right, `table == None` should return False.

Reply:
Behaviour is back to normal now.

Reply:
Oh, I just saw your edit: `table.equals(None) can still raise TypeError though.`

Are you okay with this? This is how it was before.

```
>>> import pyarrow
>>> table = pyarrow.chunked_array([], type=pyarrow.int32())
>>> table.equals(None)
False
```